libsolidity/analysis/ImmutableValidator.h

```diff
@@ -66,17 +66,21 @@ class ImmutableValidator: private ASTConstVisitor
 
 	void visitCallableIfNew(Declaration const& _declaration);
 
+	bool writtenInCurrentControlFlow(VariableDeclaration const& _variableReference) const;
+
 	ContractDefinition const& m_mostDerivedContract;
 
 	CallableDeclarationSet m_visitedCallables;
 
-	std::set<VariableDeclaration const*, ASTNode::CompareByID> m_initializedStateVariables;
+	VariableDeclarationSet m_initializedStateVariables;
+	VariableDeclarationSet m_stateVariablesNotInitializedInBothBranches;
```

Comment:
Wouldn't `m_stateVariablesInitializedInOnlyOneBranch` be a better name?

Reply:
(Variables not initialized in any of the branches are not part of the set)

Reply:
Good call, done.